lib/compat/wordpress-6.8/class-gutenberg-rest-post-search-handler.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+add_filter( 'wp_rest_search_handlers', 'gutenberg_rest_post_search_handler' );
+
+class Gutenberg_REST_Post_Search_Handler extends WP_REST_Post_Search_Handler {
+	public function prepare_item( $id, $fields ) {
+		$data = parent::prepare_item( $id, $fields );
+		if ( (int) get_option( 'page_on_front' ) === $id ) {
+			$data['is_front_page'] = true;
+		}
+		if ( (int) get_option( 'page_for_posts' ) === $id ) {
+			$data['is_blog_home'] = true;
+		}
+		return $data;
+	}
+}
+
+function gutenberg_rest_post_search_handler( $handlers ) {
+	$handlers[] = new Gutenberg_REST_Post_Search_Handler();
```

Comment:
Considering how the search controller will subsequently initialise with the given handlers:

https://github.com/WordPress/wordpress-develop/blob/b1b57613a0c9cd673a436fd904092db07e6a7d83/src/wp-includes/rest-api/endpoints/class-wp-rest-search-controller.php#L69-L81

… isn't it a bit fragile to merely append a new handler and hope that the search controller's constructor will iterate over the handlers array in a specific order?

This, which I haven't tested, seems more robust and intentional:

```suggestion
	foreach ( $handlers as $i => $handler ) {
		if ( 'post' === $handler->get_type() ) {
			unset( $handlers[ $i ] );
		}
	}

	$handlers[] = new Gutenberg_REST_Post_Search_Handler();
```